packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -317,21 +302,19 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> {
 		// Compute the net change to this branch
 		const sourceCommits = rebaseResult.commits.sourceCommits;
 		assert(hasSome(sourceCommits), 0xa86 /* Expected source commits in non no-op merge */);
-		const change = this.changeFamily.rebaser.compose(sourceCommits);
-		const taggedChange = makeAnonChange(change);
+		const { rebaser } = this.changeFamily;
 		const changeEvent = {
 			type: "append",
 			kind: CommitKind.Default,
 			get change(): TaggedChange<TChange> {
-				return taggedChange;
+				return makeAnonChange(rebaser.compose(sourceCommits));
```

Comment:
This will recompute the composition on every read of `change`. I'm not sure that matters right now, but it's probably worth defending against this in case it matters (now or later)

Reply:
Good point. If we went to the trouble to make it lazy, we may as well make it cached too.

Reply:
I added a helper function for defining lazy+cached properties. We can use it in the rebase code too (and probably other places, though I don't know off the top of my head).

Reply:
I love it! Thanks for doing that.